include/timer.h

```diff
@@ -62,4 +80,61 @@ static inline void Delay(int milliseconds)
 	std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
 }
 
+static inline void DelayUs(int microseconds)
+{
+	std::this_thread::sleep_for(std::chrono::microseconds(microseconds));
+}
+
+// The duration to use for precise sleep
+static constexpr int precise_delay_duration_us = 100;
+// The tolerance to allow for sleep variation
+static constexpr int precise_delay_tolerance_us = precise_delay_duration_us; 
+// The estimate of how long the sleep should take (microseconds)
+static constexpr double precise_delay_default_estimate = 5e-5;
+// Use the estimate value as the defaul mean time taken
+static constexpr double precise_delay_default_mean = 5e-5;
+
+static inline void DelayPrecise(int milliseconds)
+{
+    static double estimate = precise_delay_default_estimate;
+    static double mean = precise_delay_default_mean;
+    static double m2 = 0;
+    static int64_t count = 1;
+
+	double seconds = milliseconds / 1e3;
+
+	// sleep as long as we can, then spinlock the rest
+    while (seconds > estimate) {
+        const auto start = GetTicksUs();
+        DelayUs(precise_delay_duration_us);
+        const double observed = GetTicksUsSince(start) / 1e6;
```

Comment:
Is the ` 1e6` a conversion so observed is in milliseconds -- or is observed supposed to be some kind of fractional tolerance?  (if the prior.. a `_ms` postfix on the variable name would do the trick).

Reply:
the original code operated on seconds, so that's a conversion to seconds; added a comment to clarify.